### PR TITLE
Add signup component and functionality

### PR DESCRIPTION
**Pull request by IncredibleBlackMan**
- Create signup component
- Make signup requests to the backend

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_BASE_URL='http://localhost:8000'
```
Comment (lukorito) [inline .env]: Should be under `env-example` file

File: package.json [hunk 2]
```diff
@@ -3,13 +3,19 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "axios": "^0.19.0",
+    "dotenv": "^8.2.0",
```
Comment (lukorito) [inline package.json]: You don't need this. 
https://create-react-app.dev/docs/adding-custom-environment-variables/

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Route, Switch } from 'react-router-dom';
+import signUpContainer from './containers/signUpContainer';
+
+import './App.scss';
+
+require('dotenv').config();
```
Comment (lukorito) [inline src/App.jsx]: No need for this

File: src/components/signUpComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        const { username, email, password, confirm_password } = this.state;
+        return(
+            <div id="body" className="signup-body">
+                <div className="form-content">
+                    {success ? <Redirect to="/login.html" /> : null}
+                    <div className="logo">
+                        <img src={ logo } alt="Blogger"/>
+                    </div>
+                    <form onSubmit={this.onSignUp}>
+                        <input
```
Comment (lukorito) [inline src/components/signUpComponent.jsx]: Can be a separate reusable component `<Input/>` that accepts the name, placeholder, default value and on change value"

File: src/components/signUpComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                            id="confirm_password"
+                            placeholder="Confirm Password"
+                            onChange={this.handleChange}
+                            defaultValue={confirm_password}
+                            required
+                        />
+
+                        {loading 
+                            ? (
+                                <div className="button-loader">
```
Comment (lukorito) [inline src/components/signUpComponent.jsx:85]: Can be a separate loader component

File: package.json (hunk 2 above)
Comment (IncredibleBlackMan) [reply package.json]: On it.

File: src/components/signUpComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    handleChange = (event) => {
+        this.setState({ [event.target.name]: event.target.value});
+    }
+
+    onSignUp = (event) => {
+        event.preventDefault();
+        const { password, confirm_password } = this.state;
+        if (password === confirm_password) {
+          const { signup } = this.props;
```
Comment (lukorito) [inline src/components/signUpComponent.jsx:24]: Maybe destructure this after the password state desctructure

File: src/components/signUpComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+                        <img src={ logo } alt="Blogger"/>
+                    </div>
+                    <form onSubmit={this.onSignUp}>
+                        <Input
+                            type="text"
+                            name="username"
+                            id="username"
+                            placeholder="Username"
+                            onChange={this.handleChange}
+                            defaultValue={username}
```
Comment (lukorito) [inline src/components/signUpComponent.jsx:49]: No need to use the `defaultValue` with an onChange handler. Either use `value` with an onChange, or defaultValue with a ref. In the first instance, the component state will be controlled by React(us), in the second instance, the state is maintained by the component(DOM) (uncontrolled).
https://reactjs.org/docs/uncontrolled-components.html 
https://reactjs.org/docs/forms.html#controlled-components

**Review by lukorito — APPROVED** [review]

**Pull request merged**